x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -4,58 +4,80 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { i18n } from '@kbn/i18n';
+import { curry } from 'lodash';
 import axios, { AxiosError, AxiosResponse } from 'axios';
 import { schema, TypeOf } from '@kbn/config-schema';
 import { getRetryAfterIntervalFromHeaders } from './lib/http_rersponse_retry_header';
 import { nullableType } from './lib/nullable';
 import { isOk, promiseResult, Result } from './lib/result_type';
 import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import { ActionsConfigurationUtilities } from '../actions_config';
 
 // config definition
 enum WebhookMethods {
   POST = 'post',
   PUT = 'put',
 }
 
-export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
-
 const HeadersSchema = schema.recordOf(schema.string(), schema.string());
-
-const ConfigSchema = schema.object({
+const configSchemaProps = {
   url: schema.string(),
   method: schema.oneOf([schema.literal(WebhookMethods.POST), schema.literal(WebhookMethods.PUT)], {
     defaultValue: WebhookMethods.POST,
   }),
   headers: nullableType(HeadersSchema),
-});
+};
+const ConfigSchema = schema.object(configSchemaProps);
+type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
 
 // secrets definition
-export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
 const SecretsSchema = schema.object({
   user: schema.string(),
   password: schema.string(),
 });
 
 // params definition
-export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+type ActionParamsType = TypeOf<typeof ParamsSchema>;
 const ParamsSchema = schema.object({
   body: schema.maybe(schema.string()),
 });
 
 // action type definition
-export const actionType: ActionType = {
-  id: '.webhook',
-  name: 'webhook',
-  validate: {
-    config: ConfigSchema,
-    secrets: SecretsSchema,
-    params: ParamsSchema,
-  },
-  executor,
-};
+export function getActionType(configurationUtilities: ActionsConfigurationUtilities): ActionType {
+  return {
+    id: '.webhook',
+    name: 'webhook',
+    validate: {
+      config: schema.object(configSchemaProps, {
+        validate: configObject => {
+          const { url }: ActionTypeConfigType = configObject;
+          if (!configurationUtilities.isWhitelistedHostname(url)) {
+            return i18n.translate(
+              'xpack.actions.builtin.webhook.unwhitelistedWebhookConfigurationError',
+              {
+                defaultMessage:
+                  'an error occurred configuring webhook with unwhitelisted target url "{url}"',
+                values: {
+                  url,
+                },
+              }
+            );
+          }
+        },
+      }),
+      secrets: SecretsSchema,
+      params: ParamsSchema,
+    },
+    executor: curry(executor)(configurationUtilities),
```

Comment:
you just **had** to slip some function-itis in here, didn't ya? hahahaha

could we also curry it into the validators - changing their signature to take the new param, which would let us move those functions back to the top-level instead of as inline functions (less nesting)

Reply:
Yeah, agreed :)
done